### DISCUSSION TITLE
Can't run cypress test due to signup failure

### DISCUSSION THREAD

**Discussion by Genne23v** [nocodb/nocodb#4147] [Q&A] [answered]
For the cypress integration test, all tests fail because it can't sign up with preset username. Can you advise how I should avoid this? Thanks

<img width="309" alt="Screen Shot 2022-10-21 at 11 45 43 AM" src="https://user-images.githubusercontent.com/59734889/197236565-cbde3dd3-ace3-4088-a90f-e1621c123ce9.png">

**Comment by o1lab** [comment]
@dstala

**Comment by bcakmakoglu** [comment, marked as answer]
What I currently do to circumvent sign up when I want to test locally is changing this line

```js
// 00_pre_configurations.js
it("Admin SignUp", () => {
      cy.task("log", "This will be output to the terminal");
      loginPage.signIn(roles.owner.credentials); // <-- change this from `signUp` to `signIn` if you want to use an existing user for tests
});
```

Or - what we do in the CI/CD is spin up a fresh nocodb instance for each spec. @dstala can probably provide more info on how that's properly done :)

**Reply by Genne23v** [reply]
Thank you for the reply. I found that I was running backend in other terminal. That's why the user database didn't reset. But I think your recommendation will certainly work.
